File: Documents/MoonLaunch/moonlaunch-moonpay-main/lib/Front-end/views/push_notification_screen.dart
import 'package:flutter/material.dart';
import 'package:moon_launch/Front-end/widgets/app_background.dart';

class PushNotificationScreen extends StatelessWidget {
  const PushNotificationScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final Size mq = MediaQuery.of(context).size;

    return Scaffold(
      extendBodyBehindAppBar: true,
      extendBody: true,
      backgroundColor: Colors.black,
      body: AppBackground(
        child: SafeArea(
          child: Stack(
            children: [
              // ===== TOP BAR (Back + Logo) =====
              Padding(
                padding: EdgeInsets.symmetric(
                  horizontal: mq.width * 0.05,
                  vertical: mq.height * 0.018,
                ),
                child: Padding(
                  padding: const EdgeInsets.only(top: 15.0),
                  child: Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      // Back button (circle)
                      InkWell(
                        onTap: () => Navigator.pop(context),
                        borderRadius: BorderRadius.circular(999),
                        child: Container(
                          width: 38,
                          height: 38,
                          decoration: BoxDecoration(
                            shape: BoxShape.circle,
                            color: Colors.white.withOpacity(0.12),
                          ),
                          child: Icon(
                            Icons.arrow_back_ios_new_rounded,
                            size: 24,
                            color: Colors.white.withOpacity(0.9),
                          ),
                        ),
                      ),

                      // Logo (top right)
                      Image.asset(
                        'assets/images/moon_launch_logo.png',
                        width: mq.width * 0.28,
                        fit: BoxFit.contain,
                      ),
                    ],
                  ),
                ),
              ),

              // ===== CONTENT =====
              Align(
                alignment: Alignment.topCenter,
                child: Padding(
                  padding: EdgeInsets.only(top: mq.height * 0.14),
                  child: Column(
                    mainAxisSize: MainAxisSize.min,
                    children: [
                      // Title
                      Text(
                        'Push Notifications',
                        textAlign: TextAlign.center,
                        style: TextStyle(
                          fontFamily: 'BernardMTCondensed',
                          fontWeight: FontWeight.w400,
                          fontSize: mq.width * 0.070,
                          color: Colors.white,
                        ),
                      ),

                      SizedBox(height: mq.height * 0.011),

                      // Subtitle
                      Padding(
                        padding: EdgeInsets.symmetric(horizontal: mq.width * 0.12),
                        child: Text(
                          'Customize your alerts to stay on top of\nthe meme market.',
                          textAlign: TextAlign.center,
                          style: TextStyle(
                            fontFamily: 'Benne',
                            fontWeight: FontWeight.w400,
                            fontSize: mq.width * 0.042,
                            height: 1.2,
                            color: const Color(0xFFC9C9C9),
                          ),
                        ),
                      ),

                      SizedBox(height: mq.height * 0.035),

                      // Bell image (CENTER)
                      // ✅ NOTE: Asset name apne project ke mutabiq rakh lena
                      // Screenshot jaisa bell arrow image yahan lagay ga.
                      Image.asset(
                        'assets/images/bellimage.png',
                        height: 322,
                        width: 322,
                        fit: BoxFit.contain,
                      ),

                      SizedBox(height: mq.height * 0.06),

                      // Enable button
                      Padding(
                        padding: EdgeInsets.symmetric(horizontal: mq.width * 0.06),
                        child: InkWell(
                          onTap: () {
                            // TODO: enable push notification action
                          },
                          borderRadius: BorderRadius.circular(60),
                          child: Container(
                            height: mq.height * 0.070,
                            width: double.infinity,
                            decoration: BoxDecoration(
                              gradient: const LinearGradient(
                                colors: [Color(0xFF2A1216), Color(0xFF9A1117)],
                                begin: Alignment.centerLeft,
                                end: Alignment.centerRight,
                              ),
                              borderRadius: BorderRadius.circular(60),
                            ),
                            child: Center(
                              child: Text(
                                'Enable Push Notifications',
                                style: TextStyle(
                                  fontFamily: 'BernardMTCondensed',
                                  fontWeight: FontWeight.w400,
                                  fontSize: mq.width * 0.052,
                                  color: Colors.white,
                                ),
                              ),
                            ),
                          ),
                        ),
                      ),
                    ],
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

File: Documents/MoonLaunch/moonlaunch-moonpay-main/assets/images/PushNotificationScreen.java
import 'package:flutter/material.dart';
import 'package:moon_launch/widgets/app_background.dart';

class PushNotificationScreen extends StatelessWidget {
  const PushNotificationScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final Size mq = MediaQuery.of(context).size;

    return Scaffold(
      extendBodyBehindAppBar: true,
      backgroundColor: Colors.black,

      appBar: AppBar(
        automaticallyImplyLeading: false,
        backgroundColor: Colors.transparent,
        surfaceTintColor: Colors.transparent,
        elevation: 0,
        toolbarHeight: mq.height * 0.12,
        titleSpacing: mq.width * 0.05,
        title: Padding(
          padding: EdgeInsets.only(top: mq.height * 0.018),
          child: Row(
            children: [
              _topBackCircleButton(
                mq: mq,
                onTap: () => Navigator.pop(context),
              ),
              const Spacer(),
              Image.asset(
                'assets/images/moon_launch_logo.png',
                width: mq.width * 0.32,
                fit: BoxFit.contain,
              ),
            ],
          ),
        ),
      ),

      body: AppBackground(
        child: SafeArea(
          child: Padding(
            padding: EdgeInsets.only(
              left: mq.width * 0.06,
              right: mq.width * 0.06,
              top: mq.height * 0.02,
              bottom: mq.height * 0.06,
            ),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.center,
              children: [
                SizedBox(height: mq.height * 0.10),

                /// ✅ Title
                Text(
                  'Push Notifications',
                  textAlign: TextAlign.center,
                  style: TextStyle(
                    fontFamily: 'BernardMTCondensed',
                    fontSize: mq.width * 0.080,
                    color: Colors.white,
                    fontWeight: FontWeight.w700,
                    height: 1.05,
                    letterSpacing: 0.2,
                  ),
                ),

                SizedBox(height: mq.height * 0.015),

                /// ✅ Subtitle (2 lines)
                Text(
                  'Customize your alerts to stay on top of\n'
                  'the market',
                  textAlign: TextAlign.center,
                  style: TextStyle(
                    fontFamily: 'Benne',
                    fontSize: mq.width * 0.042,
                    color: Colors.white.withOpacity(0.75),
                    height: 1.35,
                  ),
                ),

                SizedBox(height: mq.height * 0.08),

                /// ✅ Center bell image
                Expanded(
                  child: Center(
                    child: Image.asset(
                      // ✅ yahan aap bell wali image ka asset path dein
                      'assets/images/bellimage.png',
                      width: 322,
                      hight: 322,
                      fit: BoxFit.contain,
                    ),
                  ),
                ),

                /// ✅ Bottom gradient button
                _gradientButton(
                  mq: mq,
                  text: 'Enable Push Notifications',
                  onTap: () {
                    // TODO: enable push notifications
                  },
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }

  /// Back circle button (top-left)
  Widget _topBackCircleButton({
    required Size mq,
    required VoidCallback onTap,
  }) {
    return InkWell(
      onTap: onTap,
      borderRadius: BorderRadius.circular(999),
      child: Container(
        height: mq.width * 0.11,
        width: mq.width * 0.11,
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(999),
          color: const Color(0xFFDB2519).withOpacity(0.20),
        ),
        child: Padding(
          padding: const EdgeInsets.only(right: 3),
          child: Icon(
            Icons.arrow_back_ios_new,
            color: Colors.white,
            size: mq.width * 0.045,
          ),
        ),
      ),
    );
  }

  /// Gradient button (Yellow -> Red)
  Widget _gradientButton({
    required Size mq,
    required String text,
    required VoidCallback onTap,
  }) {
    return InkWell(
      onTap: onTap,
      borderRadius: BorderRadius.circular(999),
      child: Container(
        height: mq.height * 0.075,
        width: double.infinity,
        decoration: BoxDecoration(
          gradient: const LinearGradient(
            colors: [
              Color(0xFFFFE600), // yellow
              Color(0xFFDB2519), // red
            ],
            begin: Alignment.centerLeft,
            end: Alignment.centerRight,
          ),
          borderRadius: BorderRadius.circular(999),
        ),
        child: Center(
          child: Text(
            text,
            textAlign: TextAlign.center,
            style: TextStyle(
              fontFamily: 'BernardMTCondensed',
              fontWeight: FontWeight.w600,
              color: Colors.white,
              fontSize: mq.width * 0.048,
            ),
          ),
        ),
      ),
    );
  }
}
